TYPO3.TYPO3CR/Classes/TYPO3/TYPO3CR/Migration/Service/NodeTransformation.php

```diff
@@ -32,14 +35,49 @@ class NodeTransformation
     protected $transformationConjunctions = array();
 
     /**
+     * @Flow\Inject
+     * @var \TYPO3\Eel\EelEvaluatorInterface
+     */
+    protected $eelEvaluator;
+
+    /**
+     * @Flow\Inject
+     * @var NodeFactory
+     */
+    protected $nodeFactory;
+
+    /**
+     * @Flow\InjectConfiguration("transformations.eel.defaultContext")
+     * @var array
+     */
+    protected $defaultContextConfiguration;
+
+    /**
+     * @var array
+     */
+    protected $contextVariables;
+
+    /**
+     * @return void
+     */
+    public function initializeObject()
+    {
+        if ($this->eelEvaluator instanceof \TYPO3\Flow\Object\DependencyInjection\DependencyProxy) {
```

Comment:
$this->eelEvaluator is used as a parameter in the evaluateEelExpression method which expects a EelEvaluatorInterface. If you don't make sure the proxy is activated, at the time you pass the object to the function it could still be a DependencyProxy object.
Instead of activating the dependency in the initializeObject method, we could deactivate lazy injection via annotation. Or we could do the proxy activation like now but right before the evalualeEelExpression call to prevent unneeded activation in transformations that don't use EEL. wdyt?